comms/rabbitmq.go

```diff
@@ -39,28 +43,33 @@ type rabbitMQComms struct {
 }
 
 // AdvertiseAgent will place an agent advertisement message on the message queue
-func (rmq rabbitMQComms) AdvertiseAgent(me defs.AgentAdvert) {
+func (rmq rabbitMQComms) AdvertiseAgent(me defs.AgentAdvert) error {
 
-	queue_url := fmt.Sprintf(
-		"amqp://%s:%s@%s:%s/",
+	queue_url := fmt.Sprintf("amqp://%s:%s@%s:%s/",
 		rmq.config.AMQP.User,
 		rmq.config.AMQP.Password,
 		rmq.config.AMQP.Host,
 		rmq.config.AMQP.Port,
 	)
 
+	retries := 0
 	conn, err := amqp.Dial(queue_url)
-	if err != nil {
-		log.Error(err)
-		log.Error("Failed to connect to RabbitMQ")
-		os.Exit(1)
+	for err != nil {
+		if retries > connectRetry {
+			log.Error("Failed to connect to RabbitMQ")
+			return err
+		}
+
+		retries++
+		log.Warnf("Failure connecting to RabbitMQ - retry #%d", retries)
+		time.Sleep(1 * time.Second)
```

Comment:
Oh wow, good catch! Silly me.

It looks like I didn't catch these because this function would eventually exceed the retry counter even if rmq was accessible, and for these function in particular, the loop that they're being called from would have ended up calling this function again - but there's a lot of wasted time in there by not retrying the connect inside the loop, obviously.